src/UdpHubListener.cpp

```diff
@@ -123,145 +124,151 @@ UdpHubListener::~UdpHubListener()
+    cout << "JackTrip HUB SERVER: Client Connect Received from Address : "
+         << PeerAddress.toString().toStdString() << endl;
+         
+    // Get UDP port from client
+    // ------------------------
+    QString clientName = QString();
+    cout << "JackTrip HUB SERVER: Reading UDP port from Client..." << endl;
+    if (clientConnection->bytesAvailable() < (int)sizeof(uint16_t)) {
+        // We don't have enough data. Wait for the next readyRead notification.
+        return;
+    }
+    int peer_udp_port = readClientUdpPort(clientConnection, clientName);
+    //TODO: Check if this is the best way to handle this now that we've refactored this code.
+    if ( peer_udp_port == 0 ) { return; }
+    cout << "JackTrip HUB SERVER: Client UDP Port is = " << peer_udp_port << endl;
+    
+    // Check is client is new or not
+    // -----------------------------
+    // Check if Address is not already in the thread pool
+    // check by comparing address strings (To handle IPv4 and IPv6.)
+    int id = isNewAddress(PeerAddress.toString(), peer_udp_port);
+    // If the address is not new, we need to remove the client from the pool
+    // before re-starting the connection
+
+    if (id == -1) {
+        int id_remove;
+        id_remove = getPoolID(PeerAddress.toString(), peer_udp_port);
+        // stop the thread
+        mJTWorkers->at(id_remove)->stopThread();
+        // block until the thread has been removed from the pool
+        while ( isNewAddress(PeerAddress.toString(), peer_udp_port) == -1 ) {
+            cout << "JackTrip HUB SERVER: Removing JackTripWorker from pool..." << endl;
+            QThread::msleep(10);
+        }
+        // Get a new ID for this client
+        //id = isNewAddress(PeerAddress.toIPv4Address(), peer_udp_port);
+        id = getPoolID(PeerAddress.toString(), peer_udp_port);
+    }
+    // Assign server port and send it to Client
+    int server_udp_port = mBasePort+id;
+    cout << "JackTrip HUB SERVER: Sending Final UDP Port to Client: " << server_udp_port << endl;
+    
+     if ( sendUdpPort(clientConnection, server_udp_port) == 0 ) {
+        clientConnection->close();
+        clientConnection->deleteLater();
+        releaseThread(id);
+        return;
+    }
+    
+    // Close and mark socket for deletion
+    // ----------------------------------
+    clientConnection->close();
+    clientConnection->deleteLater();
+    cout << "JackTrip HUB SERVER: Client TCP Connection Closed!" << endl;
+
+    // Spawn Thread to Pool
+    // --------------------
+    // Register JackTripWorker with the hub listener
+    delete mJTWorkers->at(id); // just in case the Worker was previously created
+    mJTWorkers->replace(id, new JackTripWorker(this, mBufferQueueLength, mUnderRunMode, clientName));
+    if (mIOStatTimeout > 0) {
+        mJTWorkers->at(id)->setIOStatTimeout(mIOStatTimeout);
+        mJTWorkers->at(id)->setIOStatStream(mIOStatStream);
+    }
+    // redirect port and spawn listener
```

Comment:
When merging with JB branch, the new JB related initialization commands from old `UdpHubListener::run` should be moved here.